### ISSUE TITLE
Python MegaCLI State regex parsing error

### ISSUE THREAD

**Issue by cinderblock** [netdata/netdata#8535]
### Bug report summary

I'm getting the same error as #4278 with latest netdata and _(afaik)_ latest MegaCLI:

> megacli: failed to parse "/usr/bin/sudo -n /sbin/megacli -LDPDInfo -aAll -NoLog" output

### WIP Edit Updates

Problem is that only under normal netdata execution, `megacli` does not connect to the controller. All other tests show that it should work...

See: https://github.com/netdata/netdata/issues/8535#issuecomment-605918310

#### netdata version
`netdata v1.20.0-262-nightly`

#### megacli version
`megacli -v`: `MegaCLI SAS RAID Management Tool  Ver 8.07.14 Dec 16, 2013`  
aka "[MegaCLI 5.5 P2](https://docs.broadcom.com/docs-and-downloads/raid-controllers/raid-controllers-common-files/8-07-14_MegaCLI.zip)" according to [Broadcom's search](https://www.broadcom.com/site-search?filters[pages][Content_Type][values][]=Downloads&per_page=100&q=megacli&sort_direction[pages]=desc&sort_field[pages]=updated_at).

##### `sudo -u netdata sudo -n /sbin/megacli -LDPDInfo -aAll -NoLog`

```txt
Adapter #0

Number of Virtual Disks: 1
Virtual Drive: 0 (Target Id: 0)
Name                :
RAID Level          : Primary-5, Secondary-0, RAID Level Qualifier-3
Size                : 8.185 TB
Sector Size         : 512
Is VD emulated      : No
Parity Size         : 2.728 TB
State               : Optimal
...
```
> Full output [`> megacli.txt`](https://github.com/netdata/netdata/files/4401645/megacli.txt) (line endings might be wrong).

#### Observations
~Looking at the regex in `megacli.chart.py`, I see that netdata is looking for `State` on the same line as `Adapter`:~
```
r'Adapter #([0-9]+) State(?:\s+)?: ([a-zA-Z]+)'
```
~In the version of MegaCLI that I'm using, it looks like they've moved the `State` a few lines down.~

~I bet a new regex would fix the problem. However, I'm sure a full solution would be a little more complicated. It would need to support both new and old MegaCLI.~

_**Update Edit**: not getting any data from MegaCLI. See: https://github.com/netdata/netdata/issues/8535#issuecomment-605899939_

~I have not yet investigated when this output style from MegaCLI changed. I guess it's also possible I'm running some different version of MegaCLI...~ _**Update Edit**: Also tried with "official" ppa: https://github.com/netdata/netdata/issues/8535#issuecomment-605972294_

##### OS / Environment
```
Linux server 5.3.0-42-generic #34-Ubuntu SMP Fri Feb 28 05:49:40 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
/etc/lsb-release:DISTRIB_ID=Ubuntu
/etc/lsb-release:DISTRIB_RELEASE=19.10
/etc/lsb-release:DISTRIB_CODENAME=eoan
/etc/lsb-release:DISTRIB_DESCRIPTION="Ubuntu 19.10"
/etc/os-release:NAME="Ubuntu"
/etc/os-release:VERSION="19.10 (Eoan Ermine)"
/etc/os-release:ID=ubuntu
/etc/os-release:ID_LIKE=debian
/etc/os-release:PRETTY_NAME="Ubuntu 19.10"
/etc/os-release:VERSION_ID="19.10"
/etc/os-release:HOME_URL="https://www.ubuntu.com/"
/etc/os-release:SUPPORT_URL="https://help.ubuntu.com/"
/etc/os-release:BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
/etc/os-release:PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
/etc/os-release:VERSION_CODENAME=eoan
/etc/os-release:UBUNTU_CODENAME=eoan
```

**Comment by ilyam8** [comment]
> I see that netdata is looking for State on the same line as Adapter

Before parsing it module finds all lines starts with `Aadapter #`, `State` and join them in a line

https://github.com/netdata/netdata/blob/5f891e554fb816c9225c4da2a8dd316ee924ed12/collectors/python.d.plugin/megacli/megacli.chart.py#L106-L109

**Comment by ilyam8** [comment]
@cinderblock `sudo -u netdata sudo -n /sbin/megacli -LDPDInfo -aAll -NoLog` output in the block formatting was a little bit broken, i tried to fix it but there is a chance i removed some lines 😅 

I need the whole output to find the problem, please check it.

**Comment by cinderblock** [comment]
> @cinderblock `sudo -u netdata sudo -n /sbin/megacli -LDPDInfo -aAll -NoLog` output in the block formatting was a little bit broken, i tried to fix it but there is a chance i removed some lines 😅

I don't see where the problem was or is. Why is my nice summary gone? 😭😭😭😭/dramaqueen

Joking aside, it is probably easier to just use this file version that is sure to be all of it: [megacli.txt](https://github.com/netdata/netdata/files/4401581/megacli.txt)
Captured with `sudo -u netdata sudo -n /sbin/megacli -LDPDInfo -aAll -NoLog > megacli.txt`.  
*Copied to Windows that might have done LF/CR conversion*

**Comment by ilyam8** [comment]
> Why is my nice summary gone?

😅 i fixed collapsible content, it is in the OP.

<img width="707" alt="Screenshot 2020-03-30 at 11 41 40" src="https://user-images.githubusercontent.com/22274335/77892556-73595680-727b-11ea-8fe5-54c2190a2590.png">

___

i will take a look what causes the parse error

**Comment by cinderblock** [comment]
@ilyam8 I've been trying to debug this myself. I'm a little confused. I'm not very familiar with Python but trying things anyway.

I've extended the error handling section slightly to see what's going on:

```python
    def check_disk_info(self):
        d = self._get_raw_data(command=self.megacli.disk_info)
        if not d:
            return False

        ads = find_adapters(d)
        if not (ads):
            self.error('failed to parse "{0}" output for adapters'.format(' '.join(self.megacli.disk_info)))
            self.error(d)
            self.error(getpass.getuser())
            self.error(ads)
            return False

        pds = find_pds(d)
        if not (pds):
            self.error('failed to parse "{0}" output for physical devices'.format(' '.join(self.megacli.disk_info)))
            return False
```

What's particularly confusing me is that `d` seems to be basically empty:
```log
2020-03-30 02:45:59: python.d ERROR: megacli[megacli] : failed to parse "/usr/bin/sudo -n /sbin/megacli -LDPDInfo -aAll -NoLog" output for adapters
2020-03-30 02:45:59: python.d ERROR: megacli[megacli] : [u'\r                                     \n', u'\n', u'Exit Code: 0x00\n']
2020-03-30 02:45:59: python.d ERROR: megacli[megacli] : netdata
2020-03-30 02:45:59: python.d ERROR: megacli[megacli] : []
2020-03-30 02:45:59: python.d INFO: plugin[main] : megacli[megacli] : check failed
```

But it's **not** entirely empty and actually has some output that certainly looks to be from `megacli`:
```log


Exit Code: 0x00
```

Interestingly, this is the output that I get when running megacli as my unprivileged user. If I run `megacli` command as `netdata` user, I get the same "empty" response. However if I run it with `sudo` under `netdata` user, it works as expected...

**Comment by cinderblock** [comment]
I'm trying use the [debugging tools](https://github.com/netdata/netdata/blob/master/collectors/python.d.plugin/README.md#how-to-debug-a-python-module) to see what's wrong.

I can `su` to `netdata` no problem.

```log
netdata@server:~$ /usr/libexec/netdata/plugins.d/python.d.plugin megacli debug trace
2020-03-30 03:24:51: python.d INFO: plugin[main] : using python v2
2020-03-30 03:24:51: python.d DEBUG: plugin[main] : looking for 'python.d.conf' in ['/etc/netdata', '/usr/lib/netdata/conf.d']
2020-03-30 03:24:51: python.d DEBUG: plugin[main] : loading '/etc/netdata/python.d.conf'
2020-03-30 03:24:51: python.d DEBUG: plugin[main] : '/etc/netdata/python.d.conf' is loaded
2020-03-30 03:24:51: python.d DEBUG: plugin[main] : looking for 'pythond-jobs-statuses.json' in /var/lib/netdata
2020-03-30 03:24:51: python.d DEBUG: plugin[main] : loading '/var/lib/netdata/pythond-jobs-statuses.json'
2020-03-30 03:24:51: python.d DEBUG: plugin[main] : '/var/lib/netdata/pythond-jobs-statuses.json' is loaded
2020-03-30 03:24:51: python.d DEBUG: plugin[main] : [megacli] looking for 'megacli.conf' in ['/etc/netdata/python.d', '/usr/lib/netdata/conf.d/python.d']
2020-03-30 03:24:51: python.d DEBUG: plugin[main] : [megacli] loading '/etc/netdata/python.d/megacli.conf'
2020-03-30 03:24:51: python.d DEBUG: plugin[main] : [megacli] '/etc/netdata/python.d/megacli.conf' is loaded
2020-03-30 03:24:51: python.d INFO: plugin[main] : [megacli] built 1 job(s) configs
2020-03-30 03:24:51: python.d DEBUG: plugin[main] : megacli[megacli] was previously active, applying recovering settings
2020-03-30 03:24:51: python.d DEBUG: megacli[megacli] : Executing command '/usr/bin/sudo -n -l'
2020-03-30 03:24:51: python.d DEBUG: megacli[megacli] : Executing command '/usr/bin/sudo -n /sbin/megacli -LDPDInfo -aAll -NoLog'
2020-03-30 03:24:51: python.d INFO: plugin[main] : megacli[megacli] : check success
CHART netdata.runtime_megacli '' 'Execution time for megacli' 'ms' 'python.d' netdata.pythond_runtime line 145000 5
DIMENSION run_time 'run time' absolute 1 1

2020-03-30 03:24:51: python.d DEBUG: megacli[megacli] : started, update frequency: 5
2020-03-30 03:24:53: python.d DEBUG: megacli[megacli] : Executing command '/usr/bin/sudo -n /sbin/megacli -LDPDInfo -aAll -NoLog'
CHART megacli.adapter_degraded '' 'Adapter State' 'is degraded' 'adapter' 'megacli.adapter_degraded' line 60000 5 '' 'python.d.plugin' 'megacli'
DIMENSION 'adapter_0_degraded' 'adapter 0' absolute 1 1 ' '

BEGIN megacli.adapter_degraded 0
SET 'adapter_0_degraded' = 0
END

CHART megacli.pd_media_error '' 'Physical Drives Media Errors' 'errors/s' 'pd' 'megacli.pd_media_error' line 60001 5 '' 'python.d.plugin' 'megacli'
DIMENSION 'slot_0_media_error' 'slot 0' incremental 1 1 ' '
DIMENSION 'slot_1_media_error' 'slot 1' incremental 1 1 ' '
DIMENSION 'slot_2_media_error' 'slot 2' incremental 1 1 ' '
DIMENSION 'slot_3_media_error' 'slot 3' incremental 1 1 ' '

BEGIN megacli.pd_media_error 0
SET 'slot_0_media_error' = 0
SET 'slot_1_media_error' = 0
SET 'slot_2_media_error' = 0
SET 'slot_3_media_error' = 0
END

CHART megacli.pd_predictive_failure '' 'Physical Drives Predictive Failures' 'failures/s' 'pd' 'megacli.pd_predictive_failure' line 60002 5 '' 'python.d.plugin' 'megacli'
DIMENSION 'slot_0_predictive_failure' 'slot 0' incremental 1 1 ' '
DIMENSION 'slot_1_predictive_failure' 'slot 1' incremental 1 1 ' '
DIMENSION 'slot_2_predictive_failure' 'slot 2' incremental 1 1 ' '
DIMENSION 'slot_3_predictive_failure' 'slot 3' incremental 1 1 ' '

BEGIN megacli.pd_predictive_failure 0
SET 'slot_0_predictive_failure' = 0
SET 'slot_1_predictive_failure' = 0
SET 'slot_2_predictive_failure' = 0
SET 'slot_3_predictive_failure' = 0
END

BEGIN netdata.runtime_megacli 0
SET run_time = 20
END

2020-03-30 03:24:53: python.d DEBUG: megacli[megacli] : update => [OK] (elapsed time: 20, failed retries in a row: 0)
```

It all seems to work fine and automatically rechecks every 5 seconds.

**Comment by ilyam8** [comment]
> Why would it work in the debug interface manually on the shell but not when running normally?

Hm, it is interesting. There is no real difference 🤔

**Comment by ilyam8** [comment]
@petarkozic as a megacli user, any ideas?

**Comment by petarkozic** [comment]
Hi @cinderblock 
please can you try to do something for testing purposes?

1. Unlink that yours a `megacli` file.
2. Install a `megacli` from repository, you have here:
[https://hwraid.le-vert.net/wiki/DebianPackages]
3. Then, check where is a megacli, need to be installed in the `/usr/sbin/megacli`
`which megacli`
4. Set sudoer file on this way:
`netdata ALL=(root)       NOPASSWD: /usr/sbin/megacli`
5. Restart the netdata

@ilyam8 I think here is some problem in binary which is used, or with sudoer permission against link file.

Thanks.

**Comment by cinderblock** [comment]
@petarkozic No change :(

----

I have confirmed the env is pretty much exactly the same between runs:
```txt
SUDO_GID=117
MAIL=/var/mail/root
USER=root
HOME=/root
SUDO_UID=111
LOGNAME=root
TERM=linux
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin
LANG=en_US.UTF-8
SUDO_COMMAND=/sbin/megacli -LDPDInfo -aAll -NoLog
SHELL=/bin/bash
SUDO_USER=netdata
LC_ALL=C
PWD=/etc/netdata
```
The only thing that changes is `TERM`.

I got this by creating a bash script at `/sbin/megacli` (replacing the symlink) that ran `env` before also running the original binary.

----

I'm really at a loss of what could be going on here.

Originally I installed MegaCLI via the `rpm` in the `zip` I linked in the first post. I used `alien` to convert it to a `deb` first.

**Comment by petarkozic** [comment]
@cinderblock 
Do you get valid response from command in linux if you run as sudo user?

`sudo megacli -LDPDInfo -aAll -NoLog`

**Comment by petarkozic** [comment]
@cinderblock 
One more question. Because debug run that mean plugin works correct.

Did you set this from manuals:

```
megacli is disabled by default. Should be explicitly enabled in python.d.conf.

megacli: yes```

**Comment by ilyam8** [comment]
@petarkozic if it wasnt disabled it wouldnt work even in debug mode

```sh
[ilyam-pc plugins.d]# ./python.d.plugin debug megacli
2020-03-30 16:21:48: python.d INFO: plugin[main] : using python v3
2020-03-30 16:21:48: python.d DEBUG: plugin[main] : looking for 'python.d.conf' in ['/opt/netdata/etc/netdata', '/opt/netdata/usr/lib/netdata/conf.d']
2020-03-30 16:21:48: python.d DEBUG: plugin[main] : loading '/opt/netdata/usr/lib/netdata/conf.d/python.d.conf'
2020-03-30 16:21:48: python.d DEBUG: plugin[main] : '/opt/netdata/usr/lib/netdata/conf.d/python.d.conf' is loaded
2020-03-30 16:21:48: python.d DEBUG: plugin[main] : looking for 'pythond-jobs-statuses.json' in /opt/netdata/var/lib/netdata
2020-03-30 16:21:48: python.d DEBUG: plugin[main] : loading '/opt/netdata/var/lib/netdata/pythond-jobs-statuses.json'
2020-03-30 16:21:48: python.d DEBUG: plugin[main] : '/opt/netdata/var/lib/netdata/pythond-jobs-statuses.json' is loaded
2020-03-30 16:21:48: python.d INFO: plugin[main] : [megacli] is disabled by default, skipping it
2020-03-30 16:21:48: python.d INFO: plugin[main] : no jobs to run
```

___

> Hm, it is interesting. There is no real difference 🤔

Really, it is either executed by netdata or by a user (debug mode) and does same work - collects data and writes to stdout. The only possible difference is user (that is why you say to switch to `netdata` user before debugging).

**Comment by cinderblock** [comment]
@petarkozic `wc -l` counts number of lines of output. Many means it worked. 3 means it did not.

```txt
cameron@server:~$ sudo -u netdata megacli -LDPDInfo -aAll -NoLog | wc -l
3
cameron@server:~$ sudo -u netdata sudo megacli -LDPDInfo -aAll -NoLog | wc -l
235
cameron@server:~$ sudo megacli -LDPDInfo -aAll -NoLog | wc -l
235
cameron@server:~$ megacli -LDPDInfo -aAll -NoLog | wc -l
3
cameron@server:~$ grep -v '#' /etc/netdata/python.d.conf
enabled: yes
default_run: no
megacli: yes
apache_cache: no
chrony: no
example: no
gunicorn_log: no
go_expvar: no
nginx_log: no
```

----

@ilyam8 yea, the debug test confirms configs are set as expected.

**Comment by ilyam8** [comment]
show 

> ps faxu | grep [n]etdata

**Comment by cinderblock** [comment]
```
cameron@server:~$ sudo ps faxu | grep [n]etdata
root     22000  0.0  0.0   8572  3888 pts/2    S    05:54   0:00              \_ sudo su - netdata -s /bin/bash
root     22002  0.0  0.0   8112  3164 pts/2    S    05:54   0:00                  \_ su - netdata -s /bin/bash
netdata  22003  0.0  0.0   8480  4424 pts/2    S+   05:54   0:00                      \_ -bash
netdata  23501  0.7  0.1 246836 42764 ?        Ssl  06:02   0:13 /usr/sbin/netdata -D
netdata  23631  0.0  0.0   4084  3224 ?        S    06:02   0:01  \_ bash /usr/libexec/netdata/plugins.d/tc-qos-helper.sh 1
netdata  23635  1.1  0.0   4648  3608 ?        S    06:02   0:20  \_ /usr/libexec/netdata/plugins.d/apps.plugin 1
netdata  23646  0.0  0.0 126460 19024 ?        Sl   06:02   0:01  \_ /usr/libexec/netdata/plugins.d/go.d.plugin 1
```

_PS: I am nearly needing to go to bed. Will try to stay on this. Thank you for the fast help!_

**Comment by cinderblock** [comment]
Just after a restart:
```txt
cameron@server:~$ sudo systemctl restart netdata && sleep .3 && sudo ps faxu | grep [n]etdata
root     22000  0.0  0.0   8572  3888 pts/2    S    05:54   0:00              \_ sudo su - netdata -s /bin/bash
root     22002  0.0  0.0   8112  3164 pts/2    S    05:54   0:00                  \_ su - netdata -s /bin/bash
netdata  22003  0.0  0.0   8480  4424 pts/2    S+   05:54   0:00                      \_ -bash
netdata  25404  0.0  0.0 229588 18436 ?        Ssl  06:35   0:00 /usr/sbin/netdata -D
netdata  25531  0.0  0.0   4084  3216 ?        S    06:35   0:00  \_ bash /usr/libexec/netdata/plugins.d/tc-qos-helper.sh 1
netdata  25532  0.0  0.0  15580 12408 ?        R    06:35   0:00  \_ /usr/bin/python /usr/libexec/netdata/plugins.d/python.d.plugin 1
netdata  25533  0.0  0.0 126204 18992 ?        Rl   06:35   0:00  \_ /usr/libexec/netdata/plugins.d/go.d.plugin 1
netdata  25544  0.0  0.0 222640 31236 ?        Rl   06:35   0:00  \_ /usr/bin/nodejs /usr/libexec/netdata/plugins.d/node.d.plugin 1
netdata  25547  0.0  0.0   4216  3188 ?        S    06:35   0:00  \_ bash /usr/libexec/netdata/plugins.d/charts.d.plugin 1
netdata  25584  0.0  0.0   3112  1888 ?        S    06:35   0:00  \_ /usr/libexec/netdata/plugins.d/apps.plugin 1
```

**Comment by ilyam8** [comment]
@Ferroin any ideas? It [works in debug mode](https://github.com/netdata/netdata/issues/8535#issuecomment-605918310), but it doesnt work in normal mode. [Command returns nothing](https://github.com/netdata/netdata/issues/8535#issuecomment-605899939) like the user has no permissions to use it.

**Comment by cinderblock** [comment]
As a crazy thing to try, I also tried setting the sticky bit on `megacli` so that any user that uses get gets elevated to `root`. Unfortunately that did not work at all. It would not elevate my user as far as I could tell.

**Comment by petarkozic** [comment]
> ```
> cameron@server:~$ sudo -u netdata megacli -LDPDInfo -aAll -NoLog | wc -l
> 3
> cameron@server:~$ sudo -u netdata sudo megacli -LDPDInfo -aAll -NoLog | wc -l
> 235

By with this output, I think here is problem with user permission.
What is outpout of this command

`sudo -u netdata megacli -LDPDInfo -aAll -NoLog`

**Comment by cinderblock** [comment]
```
cameron@Beelzebub:~$ sudo -u netdata megacli -LDPDInfo -aAll -NoLog


Exit Code: 0x00
cameron@Beelzebub:~$
```
However I don't bevelie this is a problem. The normal way Netdata runs `megacli` is **with** `sudo`.

**Comment by petarkozic** [comment]
Ok, I'm checked at my system.
This is a valid output.



> ```
> cameron@sever:~$ sudo -u netdata megacli -LDPDInfo -aAll -NoLog
> 
> 
> Exit Code: 0x00
> cameron@sever:~$
> ```
> 
> However I don't believe this is a problem. The normal way Netdata runs `megacli` is **with** `sudo`.
> 
> I also have not found any way to run `megacli` without `root` (or `sudo`).

Ok, I'm checked at my system.
This is a valid output.

I don't have an idea.
Especially because of working in debugging mode.

**Comment by cinderblock** [comment]
Me too. I'm rebooting out of desperation... no change 😢😢😢😭😭😭😭😭😭😭

Time for sleep for me I think.

**Comment by cinderblock** [comment]
### I found it!

**TL;DR:** `CapabilityBoundingSet` in systemd service file is too restrictive!

I realized the only thing that was different was running under systemd. I remembered some confusion I had when I realized it could just redefine `/tmp` for a running program that I was trying to debug.

Looking into options set there, I tried commenting them out and `CapabilityBoundingSet` was causing the problem!

An `strace` for `megacli` showed a call to `clone`:
```
clone(child_stack=NULL, flags=CLONE_CHILD_CLEARTID|CLONE_CHILD_SETTID|SIGCHLD, child_tidptr=0x7fdb5f523850) = 11197
```
I noticed in http://man7.org/linux/man-pages/man7/capabilities.7.html that `clone()`  requires `CAP_SYS_ADMIN` in some circumstances. Adding that to the list, it works!

Looking at the docs for `CAP_SYS_ADMIN`, I see that it is a very permissive option. It might be nice to not use such a permissive capability but I'm not sure I want to spend the time to figure out which one.

Note, because it is so permissive, I may have just gotten lucky and it's some other permission of `CAP_SYS_ADMIN` that is actually working.

#### One line fix
```
(echo '[Service]'; echo CapabilityBoundingSet=CAP_SYS_ADMIN) | sudo tee /etc/systemd/system/netdata.service.d/cap-sys-admin.conf
```
(And reboot netdata: `sudo systemctl restart netdata`)

**Comment by ilyam8** [comment]
It is likely i dont understand the option, but according name it doesnt give you capabilities it adds a list of capabilities that can be obtained (via parent process for example).

> https://www.freedesktop.org/software/systemd/man/systemd.exec.html#Capabilities

> If this option is not used, the capability bounding set is not modified on process execution, hence no limits on the capabilities of the process are enforced. 

> CapabilityBoundingSet in systemd service file is too restrictive!

i checked (`systemctl cat netdata.service`), there is no `CapabilityBoundingSet` 🤔

**Comment by cinderblock** [comment]
@ilyam8 Yeah, I think it's a systemd file that was left from when I first installed netdata with `sudo apt install netdata`. That was I think `v1.9.0`.

I see a different server of mine setup solely with the `bash < ...` script that has a service file that does not have the `CapabilityBoundingSet`.

My guess is that the install script did not replace the existing service file.

Is there a good way to force it to reinstall?

**Comment by cinderblock** [comment]
#773 talks about this setting...

**Comment by ilyam8** [comment]
> Is there a good way to force it to reinstall?

@prologic

**Comment by prologic** [comment]
> > Is there a good way to force it to reinstall?
> 
> @prologic

How did you install NetData? Looks like from source with nightly updates?

if so I'd just rebuild from source and drop in the auto-generated `system/netdata.*service` into your SystemD configuration and `systemctl daemon-reload`

**Comment by cinderblock** [comment]
@prologic I think from default ubuntu apt package. I thought I'd `apt remove`d that one before installing with the nightly update script. Not sure how that file stuck around. I'm half worried that others are still there.

Is there a way to force the install script to replace the service file? Maybe just removing it and running it again?

**Comment by cinderblock** [comment]
# Final Update

I think I figured out what the real problem was:

The previous package installation left behind:  
`/etc/systemd/system/netdata.service`  
while netdata installed:  
`/lib/systemd/system/netdata.service`  
and I didn't notice the discrepancy in `systemctl status netdata`...

Unfortunately no way for me to check completely now.

I wonder if it would be worth putting an extra check/warning/deletion in the netdata installer just for this problematic potential leftover `.service` file.

**Comment by ilyam8** [comment]
@cinderblock it is amazing how you are persistent in solving this problem 😄 👍

**Comment by cinderblock** [comment]
@ilyam8 Thanks! (Just updated my findings too lol)

I just hate coming across old answers online that were solved but no one posts the solution. Doing my part to help prevent that :D
